File: javascript/javascript-sdk/test_javascript_sdk/CommonTestSetup.js
import KestraClient from "../src/KestraClient";
import * as path from "node:path";
import {readFileSync} from "node:fs";

export const host = "http://localhost:9903"
export const username = "root@root.com"
export const password = "Root!1234"
export const MAIN_TENANT = "main";

export function kestraClient() {
    return new KestraClient(host, null, username, password);
}
export function randomId() {
    return Math.random().toString(36).substring(2, 10);
}

export function randomIdWith(str) {
    return Math.random().toString(36).substring(2, 10) + str;
}

export function randomEmail() {
    return randomId() + '@example.com';
}

export const TEST_DATA_PATH = '../../test-utils';

export function get(filePath) {
    const absolute = path.isAbsolute(filePath)
        ? filePath
        : path.resolve(process.cwd(), filePath);
    return readFileSync(absolute, 'utf8');
}

export function getCompleteFlow() {
    const raw = get(path.join(TEST_DATA_PATH, 'flows', 'flow_complete.yml'));
    return raw
    .split('flow_complete').join(randomId())
    .split('tests').join(randomId());
}

export function getSimpleFlow() {
    return getSimpleFlowAndId().flowBody;
}

export function getSimpleFlowAndId() {
    const flowId = randomId();
    const namespace = randomId();
    const raw = get(path.join(TEST_DATA_PATH, 'flows', 'simple_flow.yml'));

    const flowBody = raw
    .split('simple_flow_id_to_replace_by_random_id').join(flowId)
    .split('simple_flow_namespace_to_replace_by_random_id').join(namespace);

    return { flowBody, flowNamespace: namespace, flowId };
}

File: go/go-sdk/test_go_sdk/common_test_setup.go
package main

import (
	"context"
	"crypto/rand"
	"encoding/hex"
	"os"
	"path/filepath"
	"strings"

	openapiclient "github.com/kestra-io/client-sdk/go/go-sdk"
)

const (
	HOST           = "http://localhost:9904"
	MAIN_TENANT    = "main"
	TEST_DATA_PATH = "../../../test-utils"
)

func KestraTestApiClient() *openapiclient.APIClient {
	configuration := openapiclient.NewConfiguration()

	url := HOST

	//configuration.Debug = true

	configuration.Servers = []openapiclient.ServerConfiguration{
		{
			URL: url,
		},
	}

	apiClient := openapiclient.NewAPIClient(configuration)

	return apiClient
}
func GetAuthContext() context.Context {
	username := "root@root.com"
	password := "Root!1234"

	ctx := context.Background()

	basicAuth := openapiclient.BasicAuth{
		UserName: username,
		Password: password,
	}
	ctx = context.WithValue(ctx, openapiclient.ContextBasicAuth, basicAuth)
	return ctx
}
func randomId() string {
	b := make([]byte, 16)
	_, err := rand.Read(b)
	if err != nil {
		panic(err)
	}
	return hex.EncodeToString(b)
}

type FlowBodyAndId struct {
	FlowBody      string
	FlowNamespace string
	FlowId        string
}

func GetCompleteFlow() string {
	s := get(filepath.Join(TEST_DATA_PATH, "flows", "flow_complete.yml"))
	s = strings.ReplaceAll(s, "flow_complete", randomId())
	s = strings.ReplaceAll(s, "tests", randomId())
	return s
}

func GetSimpleFlow() string {
	return GetSimpleFlowAndId().FlowBody
}

func GetSimpleFlowAndId() FlowBodyAndId {
	flowId := randomId()
	namespace := randomId()
	body := get(filepath.Join(TEST_DATA_PATH, "flows", "simple_flow.yml"))
	body = strings.ReplaceAll(body, "simple_flow_id_to_replace_by_random_id", flowId)
	body = strings.ReplaceAll(body, "simple_flow_namespace_to_replace_by_random_id", namespace)
	return FlowBodyAndId{
		FlowBody:      body,
		FlowNamespace: namespace,
		FlowId:        flowId,
	}
}

func get(filePath string) string {
	if !filepath.IsAbs(filePath) {
		if wd, err := os.Getwd(); err == nil {
			filePath = filepath.Join(wd, filePath)
		}
	}
	b, err := os.ReadFile(filePath)
	if err != nil {
		panic(err)
	}
	return string(b)
}
